components/script/dom/window.rs

```diff
@@ -1143,6 +1175,31 @@ impl Window {
             self.emit_timeline_marker(marker.end());
         }
 
+        let pending_images = self.layout_rpc.pending_images();
+        for image in pending_images {
+            let id = image.id;
+            let js_runtime = self.js_runtime.borrow();
+            let js_runtime = js_runtime.as_ref().unwrap();
+            let node = from_untrusted_node_address(js_runtime.rt(), image.node);
```

Comment:
Script is paused during layout, and this executes immediately after the script thread resumes.